x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculator.java

```diff
@@ -107,18 +150,11 @@ private static long allowedBytesForMl(long machineMemory, Long jvmSize, int maxM
             if (machineMemory - jvmSize < OS_OVERHEAD || machineMemory == 0) {
                 return machineMemory / 100;
             }
-            // This calculation is dynamic and designed to maximally take advantage of the underlying machine for machine learning
-            // We only allow 200MB for the Operating system itself and take up to 90% of the underlying native memory left
-            // Example calculations:
-            // 1GB node -> 41%
-            // 2GB node -> 66%
-            // 16GB node -> 87%
-            // 64GB node -> 90%
```

Comment:
The comment might be useful in the future.  I can see that it hardcodes an assumption about the OS overhead constant, but instead of deleting it it could be rephrased as "if the OS overhead were 200MB then these are some example results".